script/functions.sh

```diff
@@ -16,6 +16,11 @@ if is_mri_192_plus; then
   export RUBYOPT="--disable=gem"
 fi
 
+if is_ruby_34_plus; then
+  export RUBYOPT="$RUBYOPT -W:deprecated"
+  export RUBYOPT="$RUBYOPT --enable=frozen-string-literal --debug=frozen-string-literal"
+fi
```

Comment:
I think this needs to be applied only to our specs

Reply:
It’s the purpose to apply to everything:

> This is going to be a community effort 💪.

> To help in this transition, you can enable warnings in CI and note which gems issue warnings. Then, report them to the gem maintainers.

Reply:
Our CI should not be failing because other gems raise warnings, in particular I think this is why the yard check is failing.

Reply:
Well, the intention wasn’t to fail our 3.4 build job when any of the (including development) dependencies mistakenly introduces a string literal mutation. But rather have such a build job as optional. If we’d observe a red one, that would tell us to check if it’s coming from the PR changes or from dependencies.
But it wouldn’t prevent from merging if it’s not our code.
For that reason, I planned to add two 3.4 build jobs, one with this check, and one conventional, required one.

A justification for this is to help the ecosystem.
We can probably even add this build yo `main` only and skip for PRs.

Reply:
I want all checks on our repos to pass because my experience is that having a failing build, even just ruby-head, becomes to easy to ignore when there are actual failures "oh it always fails so its fine <merge> syndrome".

If you want to have something that looks for warnings in the wider ecosystem, I'd recommend setting something up with a cron action in your own repo away from rspec